main/src/library/System.flix

```diff
@@ -30,28 +30,28 @@ namespace System {
         ///
         /// See also `Console.readLine` for reading from the console.
         ///
-        pub def readLines(): Iterator[String, Static] \ IO =
+        pub def readLines(r: Region[r]): Iterator[String, r] \ { Read(r), Write(r), IO } =
             import static get java.lang.System.in: ##java.io.InputStream \ IO as getSystemIn;
             import new java.io.InputStreamReader(##java.io.InputStream): ##java.io.InputStreamReader \ IO as newInputStream;
             import new java.io.BufferedReader(##java.io.Reader): ##java.io.BufferedReader \ IO as newBufferedReader;
             // note: BufferedReader.readLine can block
             import java.io.BufferedReader.readLine(): String \ IO as brReadLine;
 
             let br = newBufferedReader(upcast newInputStream(getSystemIn())); // up-cast to superclass
-            let nextLine = ref None;
+            let nextLine = ref None @ r;
             let done = () -> match deref nextLine {
                 case Some(line) =>
                     Object.isNull(line)
                 case None =>
-                    let n = br |> brReadLine;
+                    let n = unsafe_cast brReadLine(br) as _ \ Read(r);
```

Comment:
Are  these casts needed? I would prefer not to introduce new casts.

Reply:
This is exactly the issue we discussed [here](https://github.com/flix/flix/issues/5131#issuecomment-1362754812)

We could avoid the cast by changing the `import` line so that `brReadLine` has the effect `Read(r)` instead, but both approaches are "lies" and the more explicit cast seemed better to me (someone should come and fix this up when the effect system allows it).